src/Window.cpp

```diff
@@ -50,14 +50,21 @@ const QString Window::ACK = "ack";
 const QString Window::CRASH = "crash";
 const QString Window::INVALID = "invalid";
 
-const int Window::SPEED_SLIDER_MAX = 99;
-const int Window::SPEED_SLIDER_DEFAULT = 33;
-const double Window::PROGRESS_REQUIRED_FOR_MOVE = 100.0;
-const double Window::PROGRESS_REQUIRED_FOR_TURN = 33.33;
+const int Window::SPEED_SLIDER_MAX = 100;
+const int Window::SPEED_SLIDER_DEFAULT = 50;
+const double Window::PROGRESS_REQUIRED_FOR_MOVE     = 100.0;
+const double Window::PROGRESS_REQUIRED_FOR_HALFMOVE = 100.0 / 2;
+const double Window::PROGRESS_REQUIRED_FOR_EDGEMOVE = 100.0 * 2 / 3;
+const double Window::PROGRESS_REQUIRED_FOR_TURN     = 33.33;
+const double Window::PROGRESS_REQUIRED_FOR_RUNTURN  = 33.33 * 1.5;
+const double Window::PROGRESS_REQUIRED_FOR_BACKTURN = 33.33 * 2;
+const double Window::PROGRESS_REQUIRED_FOR_HALFTURN = 33.33 / 2;
 const double Window::MIN_PROGRESS_PER_SECOND = 10.0;
 const double Window::MAX_PROGRESS_PER_SECOND = 5000.0;
 const double Window::MAX_SLEEP_SECONDS = 0.008;
 
+boolean real=false;
```

Comment:
This is out of place. If anything, it should be a member variable (isn't that what `m_isReal` is?). If it's not used for anything, please remove it.